protostar/starknet/cheaters/utils.py

```diff
@@ -0,0 +1,24 @@
+import os
+from typing import Optional
+
+from protostar.protostar_exception import ProtostarException
+from protostar.starknet.address import Address
+from protostar.starknet.data_transformer import CairoOrPythonData
+
+
+def assert_no_expected_calls(
+    expected_calls: dict[Address, list[tuple[str, CairoOrPythonData]]],
+    fixed_address: Optional[Address] = None,
+) -> None:
+    if expected_calls:
+        msg = "expected calls not fulfilled:"
+        any_addresses_present = False
+        for address, details in expected_calls.items():
+            if fixed_address is not None and address != fixed_address:
+                continue
+            any_addresses_present = True
```

Comment:
I think it would be needed anyway, we would need to know if we had at least one iteration in the loop, because if we didn't we'd not want to raise an error.

What you were thinking about would be correct if we had the condition `if expected_calls:` in the beginning, which you pointed out as redundant. So if we remove it, we need the flag.

Reply:
[address various feedback comments](https://github.com/software-mansion/protostar/pull/1363/commits/9b066b37a326344e44dbcc224db465ed14eddb4e)